### PR TITLE
Keep light on if interacting

### PR DESCRIPTION
**Pull request by wryun**
This makes it possible to do a bunch of things without having to keep touching the light button.

I don't really see any downside with this. If you want the light to go off, just stop touching buttons.

WARNING: haven't yet tested on real hardware. Includes a minor patch to fix what I think is a bug in the simulator (without this, the watch will freeze up on transition as the buzzer causes the simulator to sleep in a state where the last animation frame was valid if the light was on).

### CODE REVIEW

**Comment by 814d3** [comment]
A nice feature, although I would at least recommend using the following mod, as it allows the LED to be used at about half the brightness level and at the same time provides better readability. This protects the LED (lifespan), the battery and your eyes ;-).
[etsy.com](https://www.etsy.com/de-en/listing/1448973768/back-light-spreader-for-casio-f-91w-a)

**Comment by Kistelini** [comment]
I really like this feature, but I think it should be an option for the options menu because it changes "default" behavior of the watch.

**Comment by wryun** [comment]
Hmm, surely by that logic everything is set in stone? (i.e. you can't change any of the default faces? Or is it just your concern that we're changing the button behaviour?)

File: watch-library/simulator/main.c [hunk 1]
```diff
@@ -89,6 +89,7 @@ void main_loop_sleep(uint32_t ms) {
     main_loop_set_sleeping(true);
     emscripten_sleep(ms);
     main_loop_set_sleeping(false);
+    animation_frame_id = ANIMATION_FRAME_ID_INVALID;
```
Comment (matheusmoreira) [inline watch-library/simulator/main.c:92]: This is a seemingly unrelated change -- what is the point of it?

**Review by matheusmoreira — APPROVED** [review]
I think this is a good feature.

File: watch-library/simulator/main.c (hunk 1 above)
Comment (matheusmoreira) [reply watch-library/simulator/main.c:92]: You're right, it's probably the simulator workaround. Now that you mentioned it, I re-read the pull request text and it makes sense.

I don't think this is causing any problems on real hardware. It's only compiled in simulator builds.

Comment (CarpeNoctem) [reply watch-library/simulator/main.c:92]: Oh, you're right - I just saw that this change is inside the simulator folder. All good, then!
